aiidalab_widgets_base/nodes.py

```diff
@@ -210,3 +264,88 @@ def find_node(self, pk):
             if getattr(node, "pk", None) == pk:
                 return node
         raise KeyError(pk)
+
+
+class _AppIcon:
+    def __init__(self, app, app_registry, path_to_root, node):
+
+        name = app["name"]
+        app_data = app_registry["apps"][name]
+        app_object = AiidaLabApp(name, app_data, AIIDALAB_APPS)
+        icon = os.path.splitext(app_object.url)[0]
+
+        # We expect it to always be a git repository
+        icon += "/master/" + app_object.metadata["logo"]
+        if "github.com" in icon:
+            icon = icon.replace("github.com", "raw.githubusercontent.com")
+            if icon.endswith(".svg"):
+                icon += "?sanitize=true"
+        self.icon = icon
```

Comment:
It would be much easier if the logo URL would simply always be a fully-qualified URL. Then there is no need for all of this guesswork here. The [AiiDAlab registry](https://github.com/aiidalab/aiidalab-registr) already does not try to guess the logo path, but simply uses the provided URLs.